File: 40kArmyBuilder/Models/Weapon.cs
namespace _40kArmyBuilder.Models
{
    public class Weapon(string Name, 
        string[] WeaponKeyword, 
        int Range, 
        string Attack, 
        int BalisticSkill, 
        int Strength, 
        int ArmorPiercing, 
        string Damage)
    {
        public string Name { get; } = Name;
        public string[] WeaponKeyword { get; } = WeaponKeyword;
        public int Range { get; } = Range;
        public string Attack { get; } = Attack;
        public int BalisticSkill { get; } = BalisticSkill;
        public int Strength { get; } = Strength;
        public int ArmorPiercing { get; } = ArmorPiercing;
        public string Damage { get; } = Damage;
    }
}

File: lib/models/weapon.dart
class Weapon {
  final String name;
  final List<String> keywords;
  final int range;
  final String attack;
  final int ballisticSkill;
  final int strength;
  final int armorPiercing;
  final String damage;

  Weapon(
    this.name, 
    this.keywords, 
    this.range, 
    this.attack, 
    this.ballisticSkill, 
    this.strength, 
    this.armorPiercing, 
    this.damage
  );

  static Weapon fromJson(Map<String, dynamic> jsonData) {
    List<String> keywords = [];
    for (String str in jsonData["keywords"]) {
      keywords.add(str);
    }
    return Weapon(
      jsonData['name'], 
      keywords, 
      jsonData['range'], 
      jsonData['attack'], 
      jsonData['ballisticSkill'], 
      jsonData['strength'], 
      jsonData['armorPiercing'], 
      jsonData['damage']
    );
  }
}